docs/tutorials/basic/android.md

```diff
@@ -0,0 +1,346 @@
+
+```proto
+  // Obtains the Features available within the given Rectangle.  Results are
+  // streamed rather than returned at once (e.g. in a response message with a
+  // repeated field), as the rectangle may cover a large area and contain a
+  // huge number of features.
+  rpc ListFeatures(Rectangle) returns (stream Feature) {}
+```
+
+- A *client-side streaming RPC* where the client writes a sequence of messages and sends them to the server, again using a provided stream. Once the client has finished writing the messages, it waits for the server to read them all and return its response. You specify a server-side streaming method by placing the `stream` keyword before the *request* type.
+
+```proto
+  // Accepts a stream of Points on a route being traversed, returning a
+  // RouteSummary when traversal is completed.
+  rpc RecordRoute(stream Point) returns (RouteSummary) {}
+```
+
+- A *bidirectional streaming RPC* where both sides send a sequence of messages using a read-write stream. The two streams operate independently, so clients and servers can read and write in whatever order they like: for example, the server could wait to receive all the client messages before writing its responses, or it could alternately read a message then write a message, or some other combination of reads and writes. The order of messages in each stream is preserved. You specify this type of method by placing the `stream` keyword before both the request and the response.
+
+```proto
+  // Accepts a stream of RouteNotes sent while a route is being traversed,
+  // while receiving other RouteNotes (e.g. from other users).
+  rpc RouteChat(stream RouteNote) returns (stream RouteNote) {}
+```
+
+Our .proto file also contains protocol buffer message type definitions for all the request and response types used in our service methods - for example, here's the `Point` message type:
+
+```proto
+// Points are represented as latitude-longitude pairs in the E7 representation
+// (degrees multiplied by 10**7 and rounded to the nearest integer).
+// Latitudes should be in the range +/- 90 degrees and longitude should be in
+// the range +/- 180 degrees (inclusive).
+message Point {
+  int32 latitude = 1;
+  int32 longitude = 2;
+}
+```
+
+
+## Generating client code
+
+Next we need to generate the gRPC client interfaces from our .proto
+service definition. We do this using the protocol buffer compiler `protoc` with
+a special gRPC Java plugin. You need to use the
+[proto3](https://github.com/google/protobuf/releases) compiler (which supports
+both proto2 and proto3 syntax) in order to generate gRPC services.
+
+The build system for this example is also part of Java gRPC itself's build. You
+can refer to the <a
+href="https://github.com/grpc/grpc-java/blob/master/README.md">README</a> and
+<a href="https://github.com/grpc/grpc-java/blob/master/examples/android/routeguide/app/build.gradle#L26">build.gradle</a> for
```

Comment:
we should reference the release branch, rather than master
https://github.com/grpc/grpc-java/blob/{{ site.dtaa.config.grpc_release_branch }}/examples...